### PR TITLE
running grouparoo, testing grouparoo, and code-config updates

### PR DESCRIPTION
**Pull request by evantahler**
This PR adds 3 new docs pages:
* `/docs/running-grouparoo` - an overview of the 3 phases of the grouparoo lifecycle
* `/docs/running-grouparoo/commands` - a page dedicated to all the grouparoo CLI commands and what they do
* `/docs/running-grouparoo/testing` - a page sharing how to use jest to test your grouparoo application

Screen Shot 2021-02-04 at 2 49 50 PM
Screen Shot 2021-02-04 at 2 49 23 PM
Screen Shot 2021-02-04 at 2 49 31 PM

Of note:
* `/docs/running-grouparoo/commands` relies on data exported from core (see https://github.com/grouparoo/grouparoo/pull/1263) and a new `CLICommandData` component to render the data
* A few updates to remove `guids` and bring the `code-config` pages up-to-date

### CODE REVIEW

File: pages/docs/running-grouparoo.mdx [hunk 1]
```diff
@@ -1,8 +1,72 @@
...
+  ];
+};
+```
+
+You can learn more about the [configuration files here](/docs/deployment/code-config).
+
+<Button href="/docs/running-grouparoo/commands#generate">
+  Learn about the Generate Command
+</Button>
+
```
Comment (seancdavis) [inline pages/docs/running-grouparoo.mdx]: We had broken out Configuration as its own section so the docs flowed like a narrative for someone setting up Grouparoo:

1. Install/Deploy
2. Configure
3. Run

This whole section was originally meant to make the assumption that you were already configured. I think it's a good idea to still _mention_ configuration, but point to the config section for doing so.

(I'm thinking you'd also put validate and apply in the config section.)

That said, reworking the config section is a whole other story that I was planning to pick up, so I'm totally fine if we keep this here for now and I pull from it for now and I move it when I pick that story up today/Monday.

I updated some notes as a result of this comment [in the doc](https://docs.google.com/document/d/1iOzxhJzRdnL4lnkNwb1Daqld1HXjy9ORU8CZmkgwm4c/edit).

File: pages/docs/running-grouparoo/commands.mdx [hunk 2]
```diff
@@ -0,0 +1,113 @@
+---
+title: "Grouparoo CLI Commands"
```
Comment (seancdavis) [inline pages/docs/running-grouparoo/commands.mdx]: Based on the way we broke up the sections, I'm now thinking the CLI should be its own section. If we're using sections in the doc as a developer workflow ...

1. Install/Deploy
2. Configure
3. Run

Then the CLI, in general, doesn't fit solely within _Run_. I'd propose we have a separate section that is all about the CLI.

**Review by seancdavis — COMMENTED** [review]
I think we should discuss this. I love the content you put here, but it doesn't really flow in the way I thought we were breaking up the sections. I have a new proposal [in the doc](https://docs.google.com/document/d/1iOzxhJzRdnL4lnkNwb1Daqld1HXjy9ORU8CZmkgwm4c/edit) as a result of this.

**Comment by evantahler** [comment]
@seancdavis I didn't get a chance to work on this today... and if you don't want to wait for me to get back, totally feel free to modify this as you see fit.  Otherwise, I'll work on this next week!

**Comment by seancdavis** [comment]
@evantahler As I started playing around with this, I found that what was confusing me is that "running grouparoo" was just supposed to be about running the application itself, but then we had installation and deployment together.

So I did some more rearranging, and I'm happy with the result. The nav is getting messier, but _structurally_ this feels really good to me. It feels like a foundation on which we can build. It should make it clearer to know where to put a new doc in the future. I kept all your content. Just broke it up among several different pages. Feel free to rearrange.

[Our planning doc](https://docs.google.com/document/d/1iOzxhJzRdnL4lnkNwb1Daqld1HXjy9ORU8CZmkgwm4c/edit) is updated with the latest structure.

Aside from the open PRs, the last pass is going to be me going through the configuration section.

**Comment by evantahler** [comment]
I updated the tests to match the new breadcrumbs

**Comment by evantahler** [comment]
I think for the purposes of this PR, the re-arrangment makes sense - merge when ready!
Let's chat about the navigation one more time holistically: 
1. there are a lot of sections now - maybe we need an accordion or something to help out visually...
2. now that you've added numbers, the `install`, `configure` and `run` sections almost read like they might want to be children of "Get Started" - ie: these are the 3 steps to getting started?

**Comment by seancdavis** [comment]
Love the ideas. I [created a story](https://www.pivotaltracker.com/story/show/176864265) to capture whatever the action may be.

**Pull request merged**